book/contributions/rec_using_inchi/recipes/rec_using_inchi.md

```diff
@@ -0,0 +1,107 @@
+
+If we can name our chemical (and this name is included in the chemical databases), you can use PubChem or ChemSpider searching features, or the CACTUS Chemical Identifier Resolver, <https://cactus.nci.nih.gov/chemical/structure>, to obtain the InChI or the InChIKey.
+
+![](../images/rec_using_inchi_fig3.png)
+
+If you can name the compound according to IUPAC systematic nomenclature, you can use the Open Parser for Systematic IUPAC nomenclature (OPSIN), to calculate (this is not a search) the InChI and the InChiKey. OPSIN can be downloaded (accessed on the command line) or used online at <https://opsin.ch.cam.ac.uk/index.html>.
+
+![](../images/rec_using_inchi_fig4.png)
+
+## Case 3: I have a MOL file for my chemical
+
+If you have a MOL or SDF file for our molecule, you can use the UniChem service, <https://www.ebi.ac.uk/unichem/> to upload the file and obtain its InChI and InChIKey.
+
+You can also use the following software that you install locally to process the MOL file:
+
+-   InChI core software, <https://www.inchi-trust.org/download-latest-inchi-standard-software/>.
+-   OpenBabel, <https://openbabel.org/wiki/Main_Page>,
+
+to convert a MOL file to InChI or InChIKey.
+
+Many common molecular drawing programs can also export and import MOL files. This way. they can also be used to convert a MOL file to an InChI or InChIKey. Make sure to check that isotope and stereochemistry information is preserved when importing and exporting MOL files.
+
+## Case 3: I have the SMILES string or the InChI for my chemical
+
+In the case you are able to write our chemical as a SMILES string or already have an InChI and just need the InChIKey for it, we may use the CACTUS Chemical Identifier Resolver, <https://cactus.nci.nih.gov/chemical/structure>.
+
+![](../images/rec_using_inchi_fig5.png)
+
+![](../images/rec_using_inchi_fig6.png)
+
+Please be aware that complex InChIs can not be interpreted by the CACTUS Chemical Identifier Resolver.
+
+Again, import and export features of the common molecular drawing packages may be useful in this case. Last but not least, databases can also help find an InChI or an InChIKey from a SMILES; provided the compound is included in the database. Remember to check the responses before taking them as correct (specially for stereochemistry and isotope distribution).
+
+## Case 4: I only have the CAS-RN for my chemical
+
+If you only have the CAS-RN for your chemical, you have to resort to a database to potentially find the InChI and InChIKey. Curated chemical databases, such as the ACS Common Chemistry service, <https://commonchemistry.cas.org/> (for common chemicals) or SciFinder, <https://scifinder.cas.org>, will be among the first options to try. Other databases like PubChem, <https://pubchem.ncbi.nlm.nih.gov>, ChemSpider, <https://chemspider.com>, Wikidata, <https://www.wikidata.org/>, or Wikipedia, <https://www.wikipedia.org/>, (quote your CAS-RN) may also work in many cases for common chemicals.
```

Comment:
Replace CAS-RN in title and text with 'CAS Registry Number'.  Link to the CAS website.